src/arviz_stats/base/dataarray.py

```diff
@@ -193,5 +195,28 @@ def kde(self, da, dims=None, circular=False, grid_len=512, **kwargs):
         out = concat((grid, pdf), dim=plot_axis)
         return out.assign_coords({"bw" if da.name is None else f"bw_{da.name}": bw})
 
+    def thin(self, da, factor="auto", dims=None):
+        """Perform thinning on DataArray input."""
+        if factor == "auto" and dims is not None:
+            warnings.warn("dims are ignored if factor is auto")
+
+        if factor == "auto":
+            n_samples = da.sizes["chain"] * da.sizes["draw"]
+            ess_ave = np.minimum(self.ess(da, method="bulk"), self.ess(da, method="tail")).mean()
```

Comment:
I think we should call ess with dims=["chain", "draw"] othereise the behaviour would be unexpected oe broken if the sample_dims rcparams is modified